src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static addAnimal(animal) {
+    if (this.alive === undefined) {
+      this.alive = [];
+    }
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    this.alive = this.alive.filter(liveAnimal => animal !== liveAnimal);
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+    Animal.addAnimal(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.addAnimal(this);
+  }
```

Comment:
```suggestion
  }
```
carnivore class doesn't need a constructor